core/src/main/java/org/apache/calcite/rel/mutable/MutableScan.java

```diff
@@ -36,11 +36,12 @@ public static MutableScan of(TableScan scan) {
   @Override public boolean equals(Object obj) {
     return obj == this
         || obj instanceof MutableScan
-        && rel.equals(((MutableScan) obj).rel);
+            && rel.getTable().getQualifiedName().equals(((MutableScan) obj).rel
+                .getTable().getQualifiedName());
   }
 
   @Override public int hashCode() {
-    return rel.hashCode();
+    return rel.getTable().getQualifiedName().hashCode();
   }
```

Comment:
Is this change expected?

Reply:
Hi, Chunwei ~
Thanks a lot looking into this ~
Actually this change is from https://github.com/apache/calcite/pull/1531/files. Since that PR is merged yet and this PR has dependency, I put it here.
Like I mentioned above, I will remove this change once PR-1531 gets merged.

Reply:
Looks like pr#1531 is merged.